### PR TITLE
[Topology.mapping] Replace all beginEdit by writeAccessor to access Loc2GlobMap

### PR DESCRIPTION
**Pull request by epernod**
Depends on #3426 

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Review by alxbilger — APPROVED** [review]

**Comment by epernod** [comment]
[ci-build][with-all-tests]

File: Sofa/GUI/Component/src/sofa/gui/component/performer/RemovePrimitivePerformer.inl [hunk 1]
```diff
@@ -370,10 +370,11 @@ bool RemovePrimitivePerformer<DataTypes>::createElementList()
             for(unsigned int i=0; i<listObject.size(); ++i) // loop on all components to find mapping (only tetra for the moment)
             {
                 sofa::core::topology::TopologicalMapping *topoMap = dynamic_cast<sofa::core::topology::TopologicalMapping *>(listObject[i]);
+                const sofa::type::vector<Index>& topoMapIndices = topoMap->Loc2GlobDataVec.getValue();
                 if (topoMap)
                 {
                     // Mapping found: 1- get surface element ID in volumique topology, 2- get volume element ID behind surface element, 3- switching all variables to volumique case
-                    unsigned int volTmp = (topoMap->getLoc2GlobVec()).getValue()[selectedElem[0]];
+                    Index volTmp = topoMapIndices[selectedElem[0]];
```
Comment (fredroy) [inline Sofa/GUI/Component/src/sofa/gui/component/performer/RemovePrimitivePerformer.inl]: Looks like this change does not belong to this PR

**Review by fredroy — CHANGES_REQUESTED** [review]

**Review by fredroy — APPROVED** [review]
